srcts/src/components/darkModeSwitch.ts

```diff
@@ -0,0 +1,298 @@
+        transition-delay: var(--speed-normal);
+      }
+
+      .sun-and-moon .moon > circle {
+        transition: transform var(--speed-fast) var(--ease-in-out-2),
+          fill var(--speed-fast) var(--ease-in-out-2);
+        transition-delay: 0s;
+      }
+
+      @supports (cx: 1) {
+        .sun-and-moon .moon > circle {
+          transition: cx var(--speed-normal) var(--ease-in-out-2);
+        }
+
+        [data-theme="dark"] .sun-and-moon .moon > circle {
+          transition: cx var(--speed-fast) var(--ease-in-out-2);
+          transition-delay: var(--speed-fast);
+        }
+      }
+
+      [data-theme="dark"] .sun-and-moon > .sun {
+        transition-delay: 0s;
+        transition-duration: var(--speed-normal);
+        transition-timing-function: var(--ease-in-out-2);
+      }
+
+      [data-theme="dark"] .sun-and-moon > .sun-beams {
+        transform: scale(0.3);
+        transition: transform var(--speed-normal) var(--ease-in-out-2),
+          opacity var(--speed-fast) var(--ease-out-1);
+        transition-delay: 0s;
+      }
+    `,
+    css`
+      :host {
+        display: inline-block;
+
+        /* We control the stroke size manually here. We don't want it getting so
+        small its not visible but also not so big it looks cartoonish */
+        --stroke-w: clamp(1px, 0.1em, 6px);
+      }
+
+      button {
+        /* This is needed to let the svg use the em sizes */
+        font-size: inherit;
+
+        /* Make sure the button is fully centered */
+        display: grid;
+        place-content: center;
+
+        /* A little bit of padding to make it easier to press */
+        padding: var(--size-xxs);
+        background: none;
+        border: none;
+        aspect-ratio: 1;
+        border-radius: 50%;
+        cursor: pointer;
+        touch-action: manipulation;
+        -webkit-tap-highlight-color: transparent;
+        outline-offset: var(--size-xxs);
+
+        /* Move down to adjust for being larger than 1em */
+        transform: translateY(var(--vertical-correction));
+        margin-block-end: var(--vertical-correction);
+      }
+
+      /*
+      button:is(:hover, :focus-visible) {
+        background: var(--surface-4);
+      }
+      */
+
+      button > svg {
+        height: var(--size);
+        width: var(--size);
+        stroke-linecap: round;
+        overflow: visible;
+      }
+
+      svg line,
+      svg circle {
+        vector-effect: non-scaling-stroke;
+      }
+    `,
+  ];
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function, @typescript-eslint/no-unused-vars
+  onChangeCallback: (x: boolean) => void = (x: boolean) => {};
+
+  // onValueChange = makeValueChangeEmitter(this, this.id);
+
+  connectedCallback(): void {
+    super.connectedCallback();
+
+    this.themeAttribute =
+      this.getAttribute("theme-attribute") || "data-shinytheme";
+
+    if (window.matchMedia("(prefers-color-scheme: dark)").matches) {
+      this.themeValue = "dark";
+    }
```

Comment:
Should this only be done if the `theme-value` attribute isn't already populated with a `mode`?

Reply:
Yeah, good point!

Reply:
5fc01d70f3304862d03c0a96078e9085ae239efb and 433c0c47313ccd9fa92bcb3ef2dce13a6e75e783